### PR TITLE
btp: use own_addr_type instead of use own address

### PR DESCRIPTION
**Pull request by mringwal**
Specifying own addr type allows to configure IUT for GAP/BROB/BCST/BV-04

### CODE REVIEW

File: pybtp/btp.py [hunk 1]
```diff
@@ -507,7 +507,7 @@ def gap_wait_for_disconnection(timeout=30):
     stack.gap.wait_for_disconnection(timeout)
 
 
-def gap_adv_ind_on(ad={}, sd={}):
+def gap_adv_ind_on(ad={}, sd={}, duration=4294967295, own_addr_type=OwnAddrType.le_identity_address):
```
Comment (michal-narajowski) [inline pybtp/btp.py]: I would create a constant value for max duration and use it here.

Comment (mringwal) [reply pybtp/btp.py]: Agree. Where to add it? (I'm still new to the codebase). defs.py seems like the correct place for this, but it is autogenerated from zepyhr sources, types.py has mostly enum style items.

Comment (michal-narajowski) [reply pybtp/btp.py]: I think types.py is fine.

Comment (mringwal) [reply pybtp/btp.py]: Ok. I've added a AdDuration class with a forever constant.

File: pybtp/types.py [hunk 2]
```diff
@@ -86,6 +86,8 @@ class AdFlags:
     sim_le_br_edr_contr = 0x08
     sim_le_br_edr_host = 0x10
 
+class AdDuration:
+    forever = 4294967295
```
Comment (michal-narajowski) [inline pybtp/types.py]: Just a nitpick: you can use hexadecimal values in Python, so I think it would be more readable if the value was 0xFFFFFFFF

**Comment by mringwal** [comment]
Good idea, thanks.  I'm mainly using C all day...

File: doc/btp_spec.txt [hunk 3]
```diff
@@ -365,9 +365,10 @@ Commands and responses:
 					Use own ID address (1 octet)
```
Comment (michal-narajowski) [inline doc/btp_spec.txt]: You should update the parameter here

File: doc/btp_spec.txt [hunk 4]
```diff
@@ -365,9 +365,10 @@ Commands and responses:
...
 		Return Parameters:	<none>
 
-		Valid Address_Type parameter values:
-					0x00 = Public
-					0x01 = Random
+		Valid Own_Addr_Type parameter values:
```
Comment (michal-narajowski) [inline doc/btp_spec.txt]: Address_Type is a different parameter, so just add a note about Own_Addr_Type below this.

Comment (mringwal) [reply doc/btp_spec.txt]: Thanks, I've missed that line.

**Comment by mringwal** [comment]
I'm back. Could this be merged onto master? Is there anything missing? Thanks!

**Comment by mringwal** [comment]
Hi again. I'll continue with the integration of our stack. Could this change be merged into master please? Anything missing? Thanks!

**Review by michal-narajowski — APPROVED** [review]
LGTM

**Comment by michal-narajowski** [comment]
Hi @mringwal, thanks for updating the PR!

**Comment by mringwal** [comment]
Thanks for review & merge.

**Pull request merged**